toolkit/tools/pkg/imagecustomizerlib/customizeboot.go

```diff
@@ -128,6 +178,31 @@ func setLinuxPath(inputGrubCfgContent string, linuxPath string) (outputGrubCfgCo
 	return outputGrubCfgContent, oldKernelPath, nil
 }
 
+// Overrides the path of the kernel binary of all the linux commands within a grub config file.
+func setLinuxPathAll(inputGrubCfgContent string, linuxPath string) (outputGrubCfgContent string, err error) {
+	quotedLinuxPath := grub.QuoteString(linuxPath)
+
+	lines, err := findLinuxLineAll(inputGrubCfgContent)
+	if err != nil {
+		return "", err
+	}
+
+	outputGrubCfgContent = inputGrubCfgContent
+	// loop from last to first so that the captured locations from
+	// findGrubCommandAll are not invalidated as reconstructing
+	// outputGrubCfgContent.
+	for i := len(lines); i > 0; i = i - 1 {
+		line := lines[i-1]
+		linuxFilePathToken := line.Tokens[1]
+		start := linuxFilePathToken.Loc.Start.Index
+		end := linuxFilePathToken.Loc.End.Index
+
+		outputGrubCfgContent = outputGrubCfgContent[:start] + quotedLinuxPath + outputGrubCfgContent[end:]
```

Comment:
This is repeated code, should we move to a helper function instead?

Reply:
Yeah, we should try to reduce the duplicated code between the singular and all variants of these functions. Feel free to get rid of the `getLinuxCommandLineArgs` function if it will make this easier.

Reply:
fixed.